wiring/src/spark_wiring_ble.cpp

```diff
@@ -883,10 +885,75 @@ class BleServiceImpl {
         return endHandle_;
     }
 
+    BleConnectionHandle& connHandle() {
+        return connHandle_;
+    }
+
+    bool& characteristicsDiscovered() {
+        return characteristicsDiscovered_;
+    }
+
+    Vector<BleCharacteristic>& characteristics() {
+        return characteristics_;
+    }
+
+    int discoverAllCharacteristics(BleService& service) {
```

Comment:
This is almost identical to the same function name inside `BleDiscoveryDelegator`, we can potentially just introduce new method in `BlePeerDeviceImpl` to discover characteristics under a given service.